Dockerfile

```diff
@@ -18,49 +19,183 @@
+LABEL org.opencontainers.image.vendor="Agnostiq"
+LABEL org.opencontainers.image.url="https://covalent.xyz"
+LABEL org.opencontainers.image.documentation="https://covalent.readthedocs.io"
+LABEL org.opencontainers.image.licenses="GNU AGPL v3"
+LABEL org.opencontainers.image.base.name="${BASE_IMAGE}"
+
+COPY --from=covalent_src $BUILDROOT/ $BUILDROOT
+
+RUN <<EOL
+  if [ ! -d $BUILDROOT/dist ] ; then
+    cd $BUILDROOT/covalent_ui/webapp
+    yarn install --network-timeout 100000
+    yarn build --network-timeout 100000
+    cd $BUILDROOT
+    python setup.py sdist
+  fi
+  python -m pip install dist/covalent-*.tar.gz
+EOL
+
+# Alias installed Covalent
+FROM covalent_${COVALENT_INSTALL_TYPE} as covalent_install
+
+# Production SDK container
+FROM base AS prod_sdk
+
+ARG BUILD_DATE
+ARG BUILD_VERSION
+
+LABEL org.opencontainers.image.title="Covalent ${COVALENT_INSTALL_TYPE}"
+LABEL org.opencontainers.image.vendor="Agnostiq"
+LABEL org.opencontainers.image.url="https://covalent.xyz"
+LABEL org.opencontainers.image.documentation="https://covalent.readthedocs.io"
+LABEL org.opencontainers.image.source="https://github.com/AgnostiqHQ/covalent"
+LABEL org.opencontainers.image.licenses="GNU AGPL v3"
+LABEL org.opencontainers.image.created="${BUILD_DATE}"
+LABEL org.opencontainers.image.version="${BUILD_VERSION}"
+LABEL org.opencontainers.image.revision="${COVALENT_COMMIT_SHA}"
+LABEL org.opencontainers.image.base.name="${BASE_IMAGE}"
+
+COPY --from=build_base /usr/bin/rsync /usr/bin/rsync
+COPY --from=build_base /usr/lib/x86_64-linux-gnu/libpopt.so.0 /usr/lib/x86_64-linux-gnu/libpopt.so.0
+
+COPY --from=covalent_install $BUILDROOT/.venv/ $INSTALLROOT/.venv
+
+RUN <<EOL
+  /usr/sbin/adduser --shell /bin/bash --disabled-password --gecos "" $USER
+  chown -R $USER:$USER $INSTALLROOT
+EOL
+
+WORKDIR $INSTALLROOT
+USER $USER
+ENV PATH=$INSTALLROOT/.venv/bin:$PATH
+
+ENTRYPOINT [ "python" ]
+
+# Production server container
+FROM prod_sdk as prod_server
+
+COPY --from=build_base /usr/bin/wget /usr/bin/wget
+COPY --from=build_base /usr/lib/x86_64-linux-gnu/libpcre2-8.so.0 /usr/lib/x86_64-linux-gnu/libpcre2-8.so.0
+COPY --from=build_base /usr/lib/x86_64-linux-gnu/libpsl.so.5 /usr/lib/x86_64-linux-gnu/libpsl.so.5
+
+ARG COVALENT_SVC_PORT=48008
+
+ENV COVALENT_SVC_PORT ${COVALENT_SVC_PORT}
 ENV COVALENT_SERVER_IFACE_ANY=1
-ENV PATH=/covalent/.venv/bin:$PATH
-EXPOSE 8080
-HEALTHCHECK CMD wget --no-verbose --tries=1 --spider http://localhost:8080 || exit 1
 
-RUN covalent config \
-  && sed -i 's|^results_dir.*$|results_dir = "/covalent/results"|' /home/ubuntu/.config/covalent/covalent.conf
+EXPOSE ${COVALENT_SERVER_PORT}
+
+HEALTHCHECK CMD wget --no-verbose --tries=1 --spider http://localhost:${COVALENT_SVC_PORT} || exit 1
+
+RUN sed -i "s#$BUILDROOT#$INSTALLROOT#" $INSTALLROOT/.venv/bin/covalent
+
+#ENTRYPOINT [ "covalent" ]
+ENTRYPOINT [ "/bin/bash" ]
+
+CMD [ "-c", "covalent start --port ${COVALENT_SVC_PORT} && tail -f /dev/null" ]
```

Comment:
we should support the DASK environment variables aka. `COVALENT_NUM_WORKERS`, `COVALENT_MEM_PER_WORKER` & `COVALENT_THREADS_PER_WORKER`, `COVALENT_DEBUG_MODE` ...

`covalent start --workers $COVALENT_NUM_WORKERS --mem-per-worker $COVALENT_MEM_PER_WORKER --threads-per-worker $COVALENT_THREADS_PER_WORKER -d`

Reply:
Agreed, let's expose all the variables here and also find a spot in documentation for them